pkg/kubelet/allocation/state/state_checkpoint.go

```diff
@@ -134,11 +139,13 @@ func (sc *stateCheckpoint) SetPodResourceAllocation(podUID types.UID, alloc map[
 }
 
 // Delete deletes allocations for specified pod
-func (sc *stateCheckpoint) Delete(podUID types.UID, containerName string) error {
+func (sc *stateCheckpoint) RemovePod(podUID types.UID) error {
 	sc.mux.Lock()
 	defer sc.mux.Unlock()
-	sc.cache.Delete(podUID, containerName)
-	return sc.storeState()
+	// Skip writing the checkpoint for pod deletion, since there is no side effect to
+	// keeping a deleted pod. Deleted pods will eventually be cleaned up by RemoveOrphanedPods.
+	// The deletion will be stored the next time a non-delete update is made.
+	return sc.cache.RemovePod(podUID)
```

Comment:
If checkpoint writes are bypassed during pod deletion, what performance gains are expected? How frequently does RemoveOrphanedPods clean up the deleted pods? Is there any information about how much would the checkpoint file size increase if a significant number of deleted pods remain uncleaned? 

Sorry, I might have missed the information in the design details. Asking these questions for personal clarification and learning.